### PR TITLE
Fixed Link components usage. Now, all links use the Backstage's Link c…

### PR DESCRIPTION
**Pull request by WarlenRodrigues**
## Hey, I just made a Pull Request!

I fixed all the Link component usage. There were some components using the MUI Link component. I turned all those into Backstage's Link component. This was mapped by @adamdmharvey at issue #8146 .

### CODE REVIEW

**Comment by adamdmharvey** [comment]
Fantastic (and quick) contrib, @WarlenRodrigues ! Thanks.  Note the pipeline failed due to a DCO check, so if you can, just may need to replay your commits using the instructions here: https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin

Thank you!

File: packages/core-components/src/layout/ErrorPage/ErrorPage.tsx
Comment (adamdmharvey) [inline packages/core-components/src/layout/ErrorPage/ErrorPage.tsx:83]: Switching from a JavaScript-based browser history back to just a path is likely not going to work for this situation? Others may have more technical suggestions on how to fix, but thing this needs some tweaks.

File: plugins/circleci/src/components/BuildsPage/lib/CITable/CITable.tsx
Comment (adamdmharvey) [inline plugins/circleci/src/components/BuildsPage/lib/CITable/CITable.tsx:172]: exclamation intentional?

**Review by adamdmharvey — CHANGES_REQUESTED** [review]
super awesome!  few quick thoughts to peek at?

**Comment by adamdmharvey** [comment]
Oops, you'll also need a changes to be added to the PR.  Super easy to do, and the instructions are here: https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets
It helps the project determine what packages to release, and automatically bumps the version based on the change types.

File: packages/core-components/src/layout/ErrorPage/ErrorPage.tsx
Comment (WarlenRodrigues) [reply packages/core-components/src/layout/ErrorPage/ErrorPage.tsx:83]: I used this ".." to go back because I noticed this is already used in other components. So I kept the pattern.
You can find one example at backstage\plugins\circleci\src\components\BuildWithStepsPage\BuildWithStepsPage.tsx

File: plugins/circleci/src/components/BuildsPage/lib/CITable/CITable.tsx
Comment (WarlenRodrigues) [reply plugins/circleci/src/components/BuildsPage/lib/CITable/CITable.tsx:172]: Yeah. This ensures the row.buildUrl exists.

Comment (WarlenRodrigues) [reply plugins/circleci/src/components/BuildsPage/lib/CITable/CITable.tsx:172]: @adamdmharvey must I remove this exclamation point yer?

File: packages/core-components/src/layout/ErrorPage/ErrorPage.tsx
Comment (WarlenRodrigues) [reply packages/core-components/src/layout/ErrorPage/ErrorPage.tsx:83]: @adamdmharvey do I adjust this or follow the pattern?

Comment (adamdmharvey) [reply packages/core-components/src/layout/ErrorPage/ErrorPage.tsx:83]: not sure; will need to leave to others or the core maintainers to check on this :) if you'd tested it manually, i'm good with it!

**Comment by adamdmharvey** [comment]
Just to ensure alignment, three outstanding tasks:

* Need to add DCO to our commits. (see above)
* Need to add a change set. (see above)
* Pipeline needs to go green; currently failing on a TSC compile.

**Review by camilaibs — APPROVED** [review]

**Comment by WarlenRodrigues** [comment]
@adamdmharvey I added the changeset and the DCO. The pipeline just disappeard. Where can I see if the compile errors have been fixed?

**Comment by freben** [comment]
Hm that's odd - what if you try to rebase to fix the conflict and force push that?

**Review by Rugvip — COMMENTED** [review]
Thank you and sorry for the delay here!

Just a few small thoughts but pretty much good to go once build troubles are sorted. It needs a rebase on top of latest master to get rid of the merge conflict

**Comment by benjdlambert** [comment]
@WarlenRodrigues is this something you want to pick up to finish?

**Comment by camilaibs** [comment]
> Thank you and sorry for the delay here!
> 
> Just a few small thoughts but pretty much good to go once build troubles are sorted. It needs a rebase on top of latest master to get rid of the merge conflict

Thanks again @WarlenRodrigues, , just to follow along, are these two things missing?
- Use back navigation rather than up one level
https://github.com/backstage/backstage/pull/8156#discussion_r759690648
- Render regardless of whether we have a link
https://github.com/backstage/backstage/pull/8156#discussion_r759691937

Are you still happy to finish them? Let us know if we can help  🙂

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
Alright I think I finished up this one - @adamdmharvey wanna give it an approval if it looks ok?

**Comment by suuus** [comment]
Thanks for your contribution @WarlenRodrigues 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**
